pkg/api/apiloader.go

```diff
@@ -81,13 +80,6 @@ func (a *Apiloader) DeserializeContainerService(contents []byte, validate, isUpd
 
 	version := m.APIVersion
 	service, err := a.LoadContainerService(contents, version, validate, isUpdate, existingContainerService)
-	if service == nil || err != nil {
```

Comment:
I don't see any AKS codepath for the `DeserializeContainerService` method. The only invocations are:

- AKS Engine deploy command
- called from `LoadContainerServiceFromFile` (which is never invoked in an AKS context)

Reply:
ignore this, we use `DeserializeContainerService` to test AKS as well as AKS Engine model deserialization

Refactored the way that we actually implement this overloading (the switch/case implementation below makes it so we don't call `a.LoadContainerService` unnecessarily in the AKS case